test/navigation/NavigationSpec.scala

```diff
@@ -61,6 +61,21 @@ class NavigationSpec extends SpecBase {
         result.url shouldBe controllers.routes.CheckYourAnswersController.onPageLoad().url
       }
 
+      s"called with $WhenDidFireOrFloodHappenPage" in new Setup {
+        val result = mainNavigator.nextPage(WhenDidFireOrFloodHappenPage, CheckMode, None)(fakeRequestWithCorrectKeysAndReasonableExcuseSet("fireOrFlood"))
+        result.url shouldBe controllers.routes.CheckYourAnswersController.onPageLoad().url
+      }
+
+      s"called with $WasHospitalStayRequiredPage" in new Setup {
```

Comment:
Could you add a test where - called with WasHospitalStayRequiredPage and answer for WhenDidHealthIssueHappenPage is set, so it goes to CYA page?